opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchScrollRequest.java

```diff
@@ -143,15 +147,50 @@ public SearchScrollRequest scrollRequest() {
    * to be reused across different physical plan.
    */
   public void reset() {
-    scrollId = null;
+    scrollId = NO_SCROLL_ID;
   }
 
   /**
    * Convert a scroll request to string that can be included in a cursor.
    * @return a string representing the scroll request.
    */
   @Override
-  public String toCursor() {
-    return scrollId;
+  public boolean hasAnotherBatch() {
+    return !needClean && !scrollId.equals(NO_SCROLL_ID);
+  }
+
+  @Override
+  public void writeTo(StreamOutput out) throws IOException {
+    initialSearchRequest.writeTo(out);
+    out.writeTimeValue(scrollTimeout);
+    out.writeBoolean(needClean);
+    if (!needClean) {
+      // If needClean is true, there is no more data to get from OpenSearch and scrollId is
+      // used only to clean up OpenSearch context.
+
+      out.writeString(scrollId);
+    }
+    out.writeStringCollection(includes);
+    indexName.writeTo(out);
+  }
+
+  /**
+   * Constructs OpenSearchScrollRequest from serialized representation.
+   * @param in stream to read data from.
+   * @param engine OpenSearchSqlEngine to get node-specific context.
+   * @throws IOException thrown if reading from input {@param in} fails.
+   */
+  public OpenSearchScrollRequest(StreamInput in, OpenSearchStorageEngine engine)
+      throws IOException {
+    initialSearchRequest = new SearchRequest(in);
+    scrollTimeout = in.readTimeValue();
+    needClean = in.readBoolean();
+    if (!needClean) {
+      scrollId = in.readString();
+    }
```

Comment:
Is it possible?

Reply:
It's a valid state of the request. 

The only reason scrollId is not serialized here is because when `needClean` is true, then scrollId will be invalid after this request is cleaned up.